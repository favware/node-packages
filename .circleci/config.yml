version: 2.1

orbs:
  codecov: codecov/codecov@1.0.5

aliases:
  - &checkout
    checkout
  - &store_artifacts
    store_artifacts:
      path: coverage
  - &codecov
    codecov/upload:
      file: coverage/*.json

run_yarn: &run_yarn
  name: Install yarn packages
  command: yarn install --frozen-lockfile

save_git: &save_git
  save_cache:
    name: Save Git Cache
    key: git-code-{{ .Revision }}
    paths:
      - .git
      - ~/.ssh

save_yarn: &save_yarn
  save_cache:
    name: Save Yarn Package Cache
    key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
    paths:
      - ~/.cache/yarn
      - node_modules

restore_git: &restore_git
  restore_cache:
    name: Restore Git Cache
    key: git-code-{{ .Revision }}

restore_yarn: &restore_yarn
  restore_cache:
    name: Restore Yarn Package Cache
    key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}

commands:
  greenkeeper_update:
    description: Updates the lockfile if current branch is a greenkeeper branch
    steps:
    - *checkout
    - <<: [*restore_git, *restore_yarn]
    - run:
        name: update lockfile
        command:
          export PATH=$PATH:$(yarn global bin) &&
          yarn global add greenkeeper-lockfile@2 &&
          greenkeeper-lockfile-update
    - run:
        name: Install dependencies
        command: yarn install --frozen-lockfile
    - <<: [*save_yarn, *save_git]
  nodejs:
    description: Primary flow
    steps:
      - *checkout
      - <<: [*restore_git, *restore_yarn]
      - run:
          <<: *run_yarn
      - run:
          name: Lint all libraries
          command: yarn lint
      - run:
          name: Test all libraries
          command: yarn ci:test
      - *store_artifacts
      - *codecov
  greenkeeper_upload:
    description: Uploads the lockfile if current branch is a greenkeeper branch
    steps:
    - *checkout
    - <<: [*restore_git, *restore_yarn]
    - run:
        name: Upload lockfile
        command:
          export PATH=$PATH:$(yarn global bin) &&
          yarn global add greenkeeper-lockfile@2 &&
          greenkeeper-lockfile-update

# List of jobs
jobs:
  lockfile_update:
    docker:
      - image: circleci/node:10
    steps:
      - greenkeeper_update
  node-v10:
    docker:
      - image: circleci/node:10
    steps:
      - nodejs
  node-v11:
    docker:
      - image: circleci/node:11
    steps:
      - nodejs
  node-v12:
    docker:
      - image: circleci/node:12
    steps:
      - nodejs
  lockfile_upload:
    docker:
      - image: circleci/node:10
    steps:
      - greenkeeper_upload

workflows:
  version: 2
  node-multi-build:
    jobs:
      - lockfile_update
      - node-v10:
          requires:
            - lockfile_update
      - node-v11:
          requires:
            - lockfile_update
      - node-v12:
          requires:
            - lockfile_update
      - lockfile_upload:
          requires:
            - node-v10
            - node-v11
            - node-v12

  nightly:
    triggers:
      - schedule:
          cron: 0 2 * * *
          filters:
            branches:
              only:
                - master
    jobs:
      - node-v10
      - node-v11
      - node-v12