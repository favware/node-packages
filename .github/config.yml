pull_request_rules:
  - name: Automatic Depfu merging
    conditions:
      - author=depfu[bot]
      - approved-reviews-by=Favna
      - base=main
      - label~=^Meta:\ Dependencies
    actions:
      merge:
        method: squash
