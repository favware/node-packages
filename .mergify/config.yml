pull_request_rules:
  - name: Automatic Depfu merging
    conditions:
      - approved-reviews-by=Favna
      - base=main
      - label~=^Meta:\ Dependencies
      - title~=Yarn\ dependencies
    actions:
      merge:
        method: squash
