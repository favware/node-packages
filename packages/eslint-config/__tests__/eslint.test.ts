import config from '../src';

describe('ESLint Config', () => {
  test('should export rules', () => {
    expect(config.root).toBe(true);
    expect(config.parser).toBe('@typescript-eslint/parser');
    expect(config).toMatchSnapshot();
  });
});
