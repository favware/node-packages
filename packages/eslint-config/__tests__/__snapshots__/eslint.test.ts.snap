// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint Config should export rules 1`] = `
Object {
  "env": Object {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": Array [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": false,
    },
    "ecmaVersion": 2019,
    "jsx": true,
    "project": "./tsconfig.eslint.json",
    "sourceType": "module",
  },
  "root": true,
  "rules": Object {
    "@typescript-eslint/consistent-type-definitions": Array [
      "error",
      "interface",
    ],
    "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/interface-name-prefix": Array [
      "error",
      Object {
        "prefixWithI": "never",
      },
    ],
    "@typescript-eslint/no-for-in-array": "warn",
    "@typescript-eslint/no-non-null-assertion": "off",
    "@typescript-eslint/no-unnecessary-qualifier": "warn",
    "@typescript-eslint/no-unnecessary-type-assertion": "warn",
    "@typescript-eslint/no-useless-constructor": "warn",
    "@typescript-eslint/prefer-for-of": "warn",
    "@typescript-eslint/prefer-includes": "warn",
    "@typescript-eslint/unified-signatures": "warn",
    "eqeqeq": Array [
      "error",
      "smart",
    ],
    "no-duplicate-imports": "error",
    "no-else-return": "warn",
    "no-eval": "error",
    "no-floating-decimal": "error",
    "no-implicit-coercion": "error",
    "no-invalid-this": "error",
    "no-iterator": "error",
    "no-lone-blocks": "error",
    "no-loop-func": "error",
    "no-return-assign": "error",
    "no-sequences": "error",
    "prefer-arrow-callback": "error",
    "prefer-template": "error",
    "yoda": "error",
  },
}
`;
